### DISCUSSION TITLE
dynamic LogicalWriter

### DISCUSSION THREAD

**Discussion by ierror** [G-Research/ParquetSharp#376] [General]
I want to loop over my schema and make the `DateTime` part of

> rowGroup.NextColumn().LogicalWriter\<DateTime>();

dynamic. 

Since my schema is dynamic and I have already defined all DataTypes for my columns in the schema, this step seems superfluous to me

Any idea how i can solve this?

**Comment by adamreeve** [comment]
You might want to implement an `ILogicalColumnWriterVisitor` so the DateTime type becomes a generic parameter. You can see one example of use of an `ILogicalColumnWriterVisitor` in the tests: https://github.com/G-Research/ParquetSharp/blob/1b8213600ccfa35a948f28bad463b28b104c4e17/csharp.test/TestLogicalTypeRoundtrip.cs#L43
and: https://github.com/G-Research/ParquetSharp/blob/1b8213600ccfa35a948f28bad463b28b104c4e17/csharp.test/LogicalValueSetter.cs#L5

Another good example is in ParquetSharp.DataFrame: https://github.com/G-Research/ParquetSharp.DataFrame/blob/c21717c434f74d18b5a86a3ba75465b8757a83bb/ParquetSharp.DataFrame/DataFrameWriter.cs#L8

**Comment by ierror** [comment]
Thank you!
